prow/config/config_test.go

```diff
@@ -29,6 +29,78 @@ import (
 	"k8s.io/test-infra/prow/kube"
 )
 
+func TestSpyglassConfig(t *testing.T) {
+	testCases := []struct {
+		name            string
+		spyglassConfig  string
+		expectedViewers map[string][]string
+		expectError     bool
+	}{
+		{
+			name: "Default: build log, metadata, junit",
+			spyglassConfig: `
+deck:
+  spyglass:
+    viewers:
+      "started.json|finished.json": ["MetadataViewer"]
+      "build-log.txt": ["BuildLogViewer"]
+      "artifacts/junit.*\\.xml": ["JUnitViewer"]
+`,
+			expectedViewers: map[string][]string{
+				"started.json|finished.json": {"MetadataViewer"},
+				"build-log.txt":              {"BuildLogViewer"},
+				"artifacts/junit.*\\.xml":    {"JUnitViewer"},
+			},
+			expectError: false,
+		},
+	}
+	for _, tc := range testCases {
+		// save the config
+		spyglassConfigDir, err := ioutil.TempDir("", "spyglassConfig")
```

Comment:
UNIT test that is focused on defaulting unset values should not interact with the filesystem

Reply:
How would you recommend getting around this? Creating a tempdir seems like a pretty reasonable operation here